gramjs/Utils.ts

```diff
@@ -1171,19 +1171,18 @@ export function  _getEntityPair(entityId, entities, cache, getInputPeer = getInp
 }
 */
 
-export function getMessageId(message: any): number | undefined {
+export function getMessageId(
+    message?: number | Api.TypeMessage | MessageIDLike
+): number | undefined {
     if (message === null || message === undefined) {
         return undefined;
-    }
-    if (typeof message == "number") {
+    } else if (typeof message === "number") {
         return message;
-    }
-
-    if (message.SUBCLASS_OF_ID === 0x790009e3) {
```

Comment:
I don't understand the change here, what if someone passes a channel. the channel has an "id" param but it's not the correct one.

Reply:
We only want to check the "id" of the object if it it's of type message.

Reply:
isn't the type guard from the message params suppose to throw error on that?
```ts
message?: number | Api.TypeMessage | MessageIDLike
```

Reply:
The only think in the `api.d.ts` file I don't see have id property is `MessageService` I converted the hash `0x790009e3` to decimals and find in the `api.d.ts` to see which class I should type the message params of the function. Am I getting the wrong idea?

Reply:
0x790009e3 = 2030045667 = MessageService | Message
MessageServices do have ID. 

The subclass check is just for that, the constructor ID with layer update and subclass doesn't

Reply:
Are you saying even after converted the bytes to class in `api.d.ts` file there is still a chance it got passed to the `message` params because it might be `MessageService | Message` but it's `SUBCLASS_OF_ID` is actually not === 0x790009e3?

Reply:
no, what I'm saying is that the check is basically doing 
if object instanceof Message or MessageService or any message type that might come in the future

Reply:
Ok so I assume when any message type come in the future we will probably need to create a class in `api.d.ts` file so it can parse the bytes into hence we will probably need to update `Api.TypeMessage` and `MessageIDLike` to include that new class. Which in turns, gonna make sure `getMessageId` will always received a correct typed message class object so it can return the id with 'message.id' right? 

Thus, my update should gives users some type guard whenever they use getMessageId make sure they are passing the desire input that the function needs. Because right now it is accepting `number | any` I'm not saying there is anything wrong with that because of the `if (message.SUBCLASS_OF_ID === 0x790009e3)` but it is not very friendly when the user try to figure out what they should pass in `getMessageId` function and also it is not type-safe.

Am I missing any cases that make the my changes to `getMessageId` a **breaking change**? Sorry, I'm kinda confused cuz to me my changes would still output the same thing it is just more type-safe.

Reply:
api.d.ts is generated automatically, 

also, I'm not talking about the changes to params. that's a good thing, I'm mostly talking about using "id" in object instead of object.subclass_of_id == since not everyone is going to use TS it's better to provide both a TS check and a runtime one. 


So in short, the `message.SUBCLASS_OF_ID === 0x790009e3` line is the only one I'd like to see back, all the other changes are good.

Reply:
Oh I see I was so confused haha. Sorry and thank you for you patient

Reply:
Done!